### PR TITLE
PGB-1343 Create utility views for basisinformatie on the analyse db, …

### PR DESCRIPTION
**Pull request by JKoetsier**
…containing references to related objects

### CODE REVIEW

File: src/gobapi/dump/to_db.py [hunk 1]
```diff
@@ -246,6 +246,90 @@ def dump_to_db(self, full_dump=False):
...
+        # Collect all necessary joins and select statements
+        joins = []
+        selects = [f'{main_alias}.*']
+
+        def ref(rel_alias: str, dst_has_states: bool):
+            return f"{rel_alias}.dst_id || '_' || {rel_alias}.dst_volgnummer" \
+                if dst_has_states else f"{rel_alias}.dst_id"
+
+        for relation in [k for k in self.model['references'].keys()]:
+            # Add a join and selects for each relation
```
Comment (Vegter) [inline src/gobapi/dump/to_db.py:271]: why the list comprenhension and not for relation in self.model['references'].keys() ?

Comment (Vegter) [reply src/gobapi/dump/to_db.py:271]: or just define relations:
[get_relation_name(GOBModel(), self.catalog_name, self.collection_name, relation) for relation in self.model['references'].keys() if relation]

File: src/gobapi/dump/to_db.py [hunk 2]
```diff
@@ -246,6 +246,90 @@ def dump_to_db(self, full_dump=False):
...
+            relation_name = get_relation_name(GOBModel(), self.catalog_name, self.collection_name, relation)
+
+            if not relation_name:
+                pass
+
+            # Determine if ManyReference and if destination has states
+            src_field = self.model['all_fields'].get(relation)
+            dst_catalog_name, dst_collection_name = src_field['ref'].split(':')
+            dst_has_states = GOBModel().has_states(dst_catalog_name, dst_collection_name)
```
Comment (Vegter) [inline src/gobapi/dump/to_db.py:280]: GOBModel().split_ref already does this job

File: src/gobapi/dump/to_db.py [hunk 3]
```diff
@@ -246,6 +246,90 @@ def dump_to_db(self, full_dump=False):
...
+            is_many = src_field['type'] == 'GOB.ManyReference'
+
```
Comment (Vegter) [inline src/gobapi/dump/to_db.py:282]: == fully_qualified_type_name(GOB.ManyReference)

File: src/gobapi/dump/to_db.py [hunk 4]
```diff
@@ -246,6 +246,90 @@ def dump_to_db(self, full_dump=False):
...
+        rel.src_id,
+        array_agg(rel.dst_id) dst_id,
+        {'rel.dst_volgnummer dst_volgnummer,' if dst_has_states else ''}
+        array_agg(rel.bronwaarde) bronwaarde,
+        array_agg({ref('rel', dst_has_states)}) "ref"
+    from rel_{relation_name} rel
+    where rel.{FIELD.DATE_DELETED} is null
+    group by rel.src_id
+) {relation} on {on}
+"""
```
Comment (Vegter) [inline src/gobapi/dump/to_db.py:301]: I suggest putting comments in all SQL statements (where appropriate, like this one) in order to make them readable as a nicely document SQL statement

File: src/tests/dump/test_to_db.py [hunk 5]
```diff
@@ -431,6 +431,109 @@ def test_dump_to_db_partial_no_source_ids_to_update(self, mock_get_entity_refs,
...
+        array_agg(rel.dst_id) dst_id,
+        
+        array_agg(rel.bronwaarde) bronwaarde,
+        array_agg(rel.dst_id) "ref"
+    from rel_relation_name_refB rel
+    where rel._date_deleted is null
+    group by rel.src_id
+) refB on refB.src_id = abbr._id and refB.src_volgnummer = abbr.volgnummer
+
+"""
```
Comment (Vegter) [inline src/tests/dump/test_to_db.py:529]: I suggest to make a compare_query method oid, that compares a query while removing comments. Someone can the add comments to queries to make them more readable withour having to update tests. Only the logic should be compared not the layout

File: src/gobapi/dump/to_db.py (hunk 1 above)
Comment (JKoetsier) [reply src/gobapi/dump/to_db.py:271]: Haha. That is actually a really good question.

File: src/gobapi/dump/to_db.py (hunk 2 above)
Comment (JKoetsier) [reply src/gobapi/dump/to_db.py:280]: Good one :)

File: src/gobapi/dump/to_db.py (hunk 3 above)
Comment (JKoetsier) [reply src/gobapi/dump/to_db.py:282]: Thanks!


.. It thought, there should be a function for this repeated pattern.. and then I forgot.. it appears I've thought that before

File: src/gobapi/dump/to_db.py (hunk 4 above)
Comment (JKoetsier) [reply src/gobapi/dump/to_db.py:301]: 👍

File: src/tests/dump/test_to_db.py (hunk 5 above)
Comment (JKoetsier) [reply src/tests/dump/test_to_db.py:542]: Done. Also added whitespace stripping to really ignore layout (as done elsewhere)

File: src/tests/dump/test_to_db.py [hunk 6]
```diff
@@ -431,6 +432,119 @@ def test_dump_to_db_partial_no_source_ids_to_update(self, mock_get_entity_refs,
...
+    def assertQueryEquals(self, expected, query):
+        def strip(query: str):
+            # Remove comments
+            stripped = re.sub(r'--.*\n', '', query)
+            # Collapse all whitespace into single spaces
+            stripped = re.sub(r'\s+', ' ', stripped)
+            # Remove leading and trailing whitespace
+            stripped = re.sub(r'^\s|\s$', '', stripped)
+
+            return stripped
```
Comment (Vegter) [inline src/tests/dump/test_to_db.py:444]: Top!!!

**Pull request merged**